### PR TITLE
Issue #2361: Allow `AbstractAction` to handle `_isAllowed()` logic

### PR DESCRIPTION
**Pull request by brideo**
Now `AbstractAction::_isAllowed()` is using a static binding, we can simply define the `ADMIN_RESOURCE` constant in our controllers.

I can see you have implemented my recommendation of a static binding on the `develop` branch. However I would like to take it one step further and define the constant in each controller, if this gets accepted I will update the dev docs to represent this.

This issue relates to:

https://github.com/magento/magento2/issues/2361

### CODE REVIEW

**Comment by brideo** [comment]
@antonkril,

In regards to defining this in XML, are you going to do this internally or is this something you'd like me to do?

We could probably use this PR to automate a script to do it for us then handle the parsing separately.

Have you discussed architecture or is this just a concept at the moment? 

Also you often reference the `ADMIN_RESOURCE` in `Blocks` and `Models`, using constants makes these easy to reference, this is something that would need to be considered.

**Comment by antonkril** [comment]
It would be great if you could do that, but first I wanted to know your opinion if we need ability to configure linking, or static linking is ok. 

Implementation would be pretty straightforward: introduce new configuration type, and use it to retrieve resources for actions.

**Comment by brideo** [comment]
Hi @antonkril,

A major pro of using constants is that it's easy to access the resource within other classes such as `Blocks` and `Models` without instantiating the configuration type.

However this is technically a configuration option so it does make sense for it to be defined as such, also it means people can now easily rewrite permissions while still being open for extension, but closed for modification just by defining a node in their configuration.

Redefining linking through configuration will take a lot longer to implement, however I think it's the better solution. This being said, static linking is still better than the current implementation and is less maintenance for developers, also the work has been done and tested.

Let me know your thoughts.

File: app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php [hunk 1]
```diff
@@ -14,6 +14,13 @@
 abstract class AbstractConfig extends \Magento\Backend\App\AbstractAction
 {
     /**
+     * Authorization level of a basic admin session
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Magento_Cms::config';
```
Comment (cspruiell) [inline app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php]: This should be 'Magento_Config::config'

File: app/code/Magento/ImportExport/Controller/Adminhtml/Import.php [hunk 2]
```diff
@@ -15,12 +15,9 @@
...
+     * Authorization level of a basic admin session
      *
-     * @return bool
+     * @see _isAllowed()
      */
-    protected function _isAllowed()
-    {
-        return $this->_authorization->isAllowed('Magento_ImportExport::import');
-    }
+    const ADMIN_RESOURCE = 'Magento_Customer::import';
```
Comment (cspruiell) [inline app/code/Magento/ImportExport/Controller/Adminhtml/Import.php]: This should be 'Magento_ImportExport::import'

**Comment by antonkril** [comment]
@brideo, agree, static linking is better than current approach. Configuration-based linking can always be added latter in a backwards-compatible way, if requested.

**Comment by brideo** [comment]
@antboiko Okay well I recommend we take this approach then we can used configuration linking in the future, this change will make the switch easier anyway.

I have updated the modules which were incorrect, sorry about that.

File: app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php [hunk 3]
```diff
@@ -60,7 +67,7 @@ public function dispatch(\Magento\Framework\App\RequestInterface $request)
     protected function _isAllowed()
     {
         $sectionId = $this->_request->getParam('section');
-        return $this->_authorization->isAllowed('Magento_Config::config')
+        return $this->_isAllowed()
```
Comment (arkadiych) [inline app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php]: seems, it can lead to circular dependency

**Comment by antonkril** [comment]
Let's merge after issues are fixed.

File: app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php (hunk 3 above)
Comment (brideo) [inline app/code/Magento/Config/Controller/Adminhtml/System/AbstractConfig.php]: Not sure how that slipped through sorry.

**Comment by brideo** [comment]
Okay I've fixed the issues that came up in code review, I've also just went over it again myself.

Happy to merge when you guys are.

**Comment by dsikkema-magento** [comment]
Internal ticket: MAGETWO-49567

**Comment by cspruiell** [comment]
Thanks for your contribution, @brideo. The changes have been merged.

**Comment by okorshenko** [comment]
@brideo your Pull Request merged to develop branch. Thank you for contribution!

**Comment by brideo** [comment]
No problem, would you like me to update the devdocs?

**Comment by antonkril** [comment]
Would be great. Thank you.

**Pull request closed without merging**